### PR TITLE
feat(combo): Auto-Combo snapshot generation/duplication in the UX

### LINKED ISSUE #10231: [Feature] Allow duplication of auto-combos, materialize the list into a snapshot for easy editing
### Problem / Use Case

Sometimes, I get this error:
```
 Error: All credentials for model kimi-k2.6 are cooling down [oc/nemotron-3-ultra-free (504)] (reset after 0s)
 Error: Stream produced no non-ping SSE event within 80000ms [oc/nemotron-3-ultra-free (504)]
 Error: All credentials for model kimi-k2.6 are cooling down [oc/nemotron-3-ultra-free (504)] (reset after 0s)
 Error: All credentials for model kimi-k2.6 are cooling down [oc/nemotron-3-ultra-free (504)] (reset after 0s)
 Error: Retry failed after 3 attempts: All credentials for model kimi-k2.6 are cooling down [oc/nemotron-3-ultra-free
 (504)] (reset after 0s)
```

Annoying, I would rather just omit oc/nemotron-3-ultra-free in this case. However auto/best-free doesn't allow editing.

### Proposed Solution

Allow users to duplicate auto combos, similar to how users can duplicate custom combos. At least the user can solve their immediate issue.

### Alternatives Considered

_No response_

### Acceptance Criteria

Duplication of auto combos should work.

### Area

Dashboard / UI

### Related Provider(s)

_No response_

### Additional Context

_No response_

### Expected Test Plan

_No response_

### PR DESCRIPTION
**Pull request by swingtempo**
## Summary

- AutoComboCatalog - each autocombo card (e.g. `auto/best-coding`) has a button to generate a snapshot of the combo so a user can see and edit it
- Upon clicking, the combo gets generated
- window scrolls to the newly created card

## Related Issues

- Closes #10231 (looks like Feature team already closed it and tracking is as an internal feature reqeusts)
- Video in discussion: https://github.com/diegosouzapw/OmniRoute/discussions/10355

### 1. Snapshot Weight Computation (`open-sse/services/autoCombo/virtualFactory.ts`)

- New function `computeSnapshotWeights()` scores candidates at combo creation time using:
  - **taskFit**: reasoning & vision-capable models score higher when taskFit is weighted
  - **stability**: richer capability profiles assumed more stable
  - **tierPriority / costInv**: premium tiers and free models get bonuses based on the weight pack
  - **latencyInv / health / quota**: neutral baseline (no runtime telemetry at snapshot time)
- Replaces hardcoded `weight: 1` in `createVirtualAutoComboFromPrepared()` with computed scores normalized to `[0, 1]`.

### 2. Duplicate Endpoint (`src/app/api/combos/duplicate/route.ts`) — NEW

- **POST `/api/combos/duplicate`** resolves an `auto/*` template (e.g. `auto/best-coding`, `auto/glm`) into a static combo snapshot.
- Reuses the same candidate resolution logic as `createVirtualAutoCombo()`.
- Normalizes weights to sum exactly 100, applies naming convention (`static-{name} copy`), and preserves the original weight pack in config.
- Returns **201** on success with the created combo object; validates input (400) and unknown templates (422).

### 3. Dashboard UI Updates

| File | Change |
|------|--------|
| `src/app/(dashboard)/dashboard/combos/page.tsx` | Added `handleComboCreated()` callback to auto-scroll newly created combo cards; wired it into `<AutoComboCatalog />`. Refactored ComboFormModal provider builder grid layout. |
| `src/app/(dashboard)/dashboard/combos/AutoComboCatalog.tsx` | Accepts optional `onComboCreated` prop for post-creation scroll behavior. |

### 4. Tests (Rule #18)

| Test File | Coverage |
|-----------|----------|
| `tests/unit/snapshot-weights.test.ts` | Capability differentiation, tier bonuses, neutral baselines, score clamping |
| `tests/unit/combos-duplicate-route.test.ts` | Auth gate, input validation (400), success response shape, weight distribution, naming convention, error sanitization |
| `tests/unit/combos-duplicate-resolution-audit.test.ts` | Audit-style checks for duplicate resolution edge cases |

### 5. Utilities & i18n

- `scripts/ad-hoc/dump-auto-combos.ts`: CLI helper to dump resolved auto-combo candidates (debug / inspection).
- ~47 locale files: minor string updates in `src/i18n/messages/*.json`.

---

## Why This Matters

| Problem | Solution |
|---------|----------|
| Auto-combos always gave equal weight (`1`) to every model, ignoring capability differences. | Snapshot weights differentiate reasoning/vision-capable models at creation time based on the active weight pack. |
| No way to "freeze" an auto-combo into a static combo for manual tweaking or versioning. | `/api/combos/duplicate` materializes any `auto/*` template as a persistent combo with preserved weights and metadata. |

---

## API Reference — Duplicate Endpoint

```http
POST /api/combos/duplicate
Authorization: Bearer <management-api-key>   (if requireLogin is true)
Content-Type: application/json

{
  "name": "auto/best-coding",    // required — any auto/* template
  "strategy": "priority"         // optional — defaults to "priority"
}
```

**Response (201):**

```json
{
  "id": "...",
  "name": "static-best-coding copy",
  "models": [
    { "model": "anthropic/claude-sonnet-4-20250514", "weight": 35 },
    { "model": "openai/gpt-4.1",                    "weight": 28 }
  ],
  "strategy": "priority",
  "config": {
    "sourceAutoCombo": "auto/best-coding",
    "weightPack": { "taskFit": 0.3, "stability": 0.25, ... }
  }
}
```

**Error codes:** `400` (missing/invalid name), `401/403` (auth), `422` (unknown template or no matching models), `500` (server error).

---

## Files Changed (51 files, +1412 / −249)

| Area | Files |
|------|-------|
| Core logic | `open-sse/services/autoCombo/virtualFactory.ts` |
| API route | `src/app/api/combos/duplicate/route.ts` *(new)* |
| Dashboard UI | `src/app/(dashboard)/dashboard/combos/page.tsx`, `AutoComboCatalog.tsx` |
| Tests | `tests/unit/snapshot-weights.test.ts` *(new)*, `tests/unit/combos-duplicate-route.test.ts` *(new)*, `tests/unit/combos-duplicate-resolution-audit.test.ts` *(new)* |
| i18n | ~47 locale files in `src/i18n/messages/` |
| Scripts | `scripts/ad-hoc/dump-auto-combos.ts` *(new)* |

---

## Testing

```bash
# Snapshot weights
node --import tsx/esm --test tests/unit/snapshot-weights.test.ts

# Duplicate route
node --import tsx/esm --test tests/unit/combos-duplicate-route.test.ts

# All unit tests
npm run test:unit
```

---

## Notes for Reviewers

NOTE! **Prettifier page.tsx impact** there are too many changes in page.tsx because Prettifier made a bunch of formatting changes. Main change is the OnComboCreated handler passed to AutoComboCatalog! Confirmed with AI that the large page.tsx changes are purely formatting.

1. **Snapshot weights are static** — they use capabilities and tier info available at combo creation time, not runtime telemetry (p95 latency, quota remaining). This is intentional; the function docstring explains the trade-off.
2. **Weight normalization**: snapshot scores are clamped to `[0, 1]`, then normalized so all model weights sum exactly to `100` in the duplicate endpoint (with remainder distributed round-robin).
3. **Error sanitization** follows Rule #12 — no raw stack traces in responses.

### CODE REVIEW

File: open-sse/services/autoCombo/virtualFactory.ts [hunk 1]
```diff
@@ -728,6 +784,7 @@ export async function createVirtualAutoComboFromPrepared(
   }
 
   const providerPool = [...new Set(effectivePool.map((c) => c.provider))];
+  const snapshotScores = computeSnapshotWeights(effectivePool, weights);
```
Comment (swingtempo) [inline open-sse/services/autoCombo/virtualFactory.ts:787]: @diegosouzapw 
Need confirmation from product team that this is desired and is a bug that was fixed. This isn't an area I'm familiar with.

File: src/app/(dashboard)/dashboard/combos/page.tsx [hunk 2]
```diff
@@ -3203,213 +3212,232 @@ function ComboFormModal({ isOpen, combo, onClose, onSave, activeProviders, combo
                 {builderSelectionMode === "step" && (
                   <>
                     <div className="grid grid-cols-1 md:grid-cols-3 gap-2 mt-3">
```
Comment (swingtempo) [inline src/app/(dashboard)/dashboard/combos/page.tsx:3214]: @diegosouzapw  These huge sections here are due to reformatting from Prettifier. I think this is automatic.

**Comment by diegosouzapw** [comment]
Deferido para revalidação do head atual. A rodada anterior registrou um hold de ownership, mas a inspeção atual não confirmou esse worktree; por favor, reconfirme o head e os checks antes da decisão final.

**Comment by diegosouzapw** [comment]
### Babysit summary (Unit Tests fast-path 4/4)

**Root cause** (from run 31853607931, job `Unit Tests fast-path (4/4)`, `tests/unit/i18n-glossary-consistency-check.test.ts` + `tests/unit/api-route-body-validation.test.ts` / t06 gate):

1. `POST /api/combos/duplicate` (`src/app/api/combos/duplicate/route.ts`) read `request.json()` and did hand-rolled `typeof` checks instead of routing through `validateBody()`/Zod — fails the `t06` "every API route reading request.json() validates it with Zod" gate (Hard Rule #7).
2. `duplicateAutoComboSnapshotMsg` in `src/i18n/messages/zh-CN.json` used the retired glossary synonym `提供商` instead of the canonical `提供者` for "provider" — fails the i18n glossary consistency regression guard (post-#8038 normalization).

**Fix** (commit `4f911f6`, authored by @swingtempo, co-authored by @diegosouzapw):
- Added `duplicateAutoComboSchema` (`src/shared/validation/schemas/combo.ts`) and routed the route body through `validateBody()`/`isValidationFailure()` (`src/shared/validation/helpers.ts`), preserving the same error message/shape for existing tests.
- Swapped `提供商` → `提供者` in the zh-CN string.

**Verification:**
- `node --import tsx/esm --test tests/unit/combos-duplicate-route.test.ts` — 10/10 pass.
- `node --import tsx/esm --test tests/unit/combos-duplicate-resolution-audit.test.ts` — 1/1 pass.
- `npx eslint src/app/api/combos/duplicate/route.ts src/shared/validation/schemas/combo.ts` — clean.
- Pushed to `swingtempo/duplicate-auto-combo2`; fresh CI run triggered (run `32438457440` at time of writing).

**Threads left unresolved (not addressed by this fix, out of scope):**
- `virtualFactory.ts` — product-confirmation question, unrelated to CI failure.
- `dashboard/combos/page.tsx` — Prettier reformatting note, unrelated to CI failure.

`dast-smoke` was previously `cancelled`/showed `fail` after a 30m timeout window in the prior run — no assertion failure observed; re-run triggered along with the rest of the fresh CI run. Not modified.

Ready for CI to finish + human review/merge.

**Comment by diegosouzapw** [comment]
⚠️ base-red inherited: #9985

New CI run (32438457440, after the fix push) shows 2 additional reds unrelated to this PR's diff — both are inherited base drift, not caused by this branch:
- **Docs Gates (fast-path)**: provider-count drift (docs say 346, live code now defines 347) + stale SVG diagrams — a base-tip content change, not touched by this PR.
- **No new ESLint warnings**: `eslintErrors: 0`, `eslintWarnings: 0` in the job's own metrics — the failure is the `quality-ratchet` step (`openapiCoverage.pct: 38.6 < baseline 39.2`), also base-tip drift per the open base-red issue #9985 ("ratchet drift ... is expected mid-cycle and is rebaselined at release — it is NOT a contributor concern").

Original target ("Unit Tests fast-path (4/4)") is fixed by commit `4f911f6` — see prior comment. Not chasing the base-red items per project policy (Hard Rule #21 / base-green protocol); they'll clear on the next base merge or release rebaseline.
